From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kirill Yatsenko <kiriyatsenko@gmail.com>
Date: Thu, 03 Apr 2026 17:00:00 +0200
Subject: dts: stm32mp15-pinctrl: Move I2C4 to PD12/PD13

On the decktrix v1 custom PCB, the PMIC (STPMIC1) I2C4 bus uses
PD12 (SCL) and PD13 (SDA) instead of the default PZ4/PZ5.

Move the i2c4 pinctrl nodes to the top of the controller section.

Signed-off-by: Kirill Yatsenko <kiriyatsenko@gmail.com>
---
 arch/arm/boot/dts/st/stm32mp15-pinctrl.dtsi | 36 ++++++++++++++-------
 1 file changed, 22 insertions(+), 14 deletions(-)

diff --git a/arch/arm/boot/dts/st/stm32mp15-pinctrl.dtsi b/arch/arm/boot/dts/st/stm32mp15-pinctrl.dtsi
index 534a03da2..9abd1125a 100644
--- a/arch/arm/boot/dts/st/stm32mp15-pinctrl.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp15-pinctrl.dtsi
@@ -25,6 +25,23 @@ pins {
 		};
 	};
 
+	i2c4_pins_a: i2c4-0 {
+		pins {
+			pinmux = <STM32_PINMUX('D', 12, AF4)>, /* I2C4_SCL */
+				 <STM32_PINMUX('D', 13, AF4)>; /* I2C4_SDA */
+			bias-disable;
+			drive-open-drain;
+			slew-rate = <0>;
+		};
+	};
+
+	i2c4_sleep_pins_a: i2c4-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX('D', 12, ANALOG)>, /* I2C4_SCL */
+				 <STM32_PINMUX('D', 13, ANALOG)>; /* I2C4_SDA */
+		};
+	};
+
 	adc1_in6_pins_a: adc1-in6-0 {
 		pins {
 			pinmux = <STM32_PINMUX('F', 12, ANALOG)>;
@@ -2831,22 +2848,6 @@ pins {
 		};
 	};
 
-	i2c4_pins_a: i2c4-0 {
-		pins {
-			pinmux = <STM32_PINMUX('Z', 4, AF6)>, /* I2C4_SCL */
-				 <STM32_PINMUX('Z', 5, AF6)>; /* I2C4_SDA */
-			bias-disable;
-			drive-open-drain;
-			slew-rate = <0>;
-		};
-	};
-
-	i2c4_sleep_pins_a: i2c4-sleep-0 {
-		pins {
-			pinmux = <STM32_PINMUX('Z', 4, ANALOG)>, /* I2C4_SCL */
-				 <STM32_PINMUX('Z', 5, ANALOG)>; /* I2C4_SDA */
-		};
-	};
 
 	i2c6_pins_a: i2c6-0 {
 		pins {
-- 
2.43.0
